src/main/java/com/iota/iri/TransactionValidator.java

```diff
@@ -46,40 +68,94 @@
+     * @param testnet true if we are in testnet mode, this caps {@code mwm} to {@value #TESTNET_MWM_CAP}
+     *                regardless of parameter input.
+     * @param mwm minimum weight magnitude: the minimal number of 9s that ought to appear in the end of the transaction
+     *            hash
+     */
     public void init(boolean testnet, int mwm) {
         setMwm(testnet, mwm);
 
         newSolidThread = new Thread(spawnSolidTransactionsPropagation(), "Solid TX cascader");
         newSolidThread.start();
     }
 
+    //Package Private For Testing
     void setMwm(boolean testnet, int mwm) {
         minWeightMagnitude = mwm;
 
         //lowest allowed MWM encoded in 46 bytes.
         if (!testnet){
-            minWeightMagnitude = Math.max(minWeightMagnitude, 13);
+            minWeightMagnitude = Math.max(minWeightMagnitude, TESTNET_MWM_CAP);
         }
     }
 
+    /**
+     * Shutdown roots to tip solidification thread
+     * @throws InterruptedException
+     * @see #spawnSolidTransactionsPropagation()
+     */
     public void shutdown() throws InterruptedException {
         shuttingDown.set(true);
         newSolidThread.join();
     }
 
+    /**
+     * @return the number of trailing 9s that have to be present in the transaction hash to validate that proof of work
+     * has been done
+     */
     public int getMinWeightMagnitude() {
         return minWeightMagnitude;
     }
 
+    /**
+     * Validates that the timestamp of the transaction is not below the last global snapshot time
+     * or more than {@value #MAX_TIMESTAMP_FUTURE} seconds in the future.
+     *
+     * <p>
+     *     First the attachment timestamp (set after performing POW) is checked, and if not available
+     *     the regular timestamp is checked. Genesis transaction will always be valid.
+     * </p>
+     * @param transactionViewModel transaction under test
+     * @return <tt>true</tt> if timestamp is not in bound and {@code transactionViewModel} is not genesis.
+     * Else returns <tt>false</tt>.
+     */
     private boolean hasInvalidTimestamp(TransactionViewModel transactionViewModel) {
         if (transactionViewModel.getAttachmentTimestamp() == 0) {
-            return transactionViewModel.getTimestamp() < snapshotTimestamp && !Objects.equals(transactionViewModel.getHash(), Hash.NULL_HASH)
+            return transactionViewModel.getTimestamp() < snapshotTimestamp
+                    //you are valid if you are the genesis
+                    && !Objects.equals(transactionViewModel.getHash(), Hash.NULL_HASH)
                     || transactionViewModel.getTimestamp() > (System.currentTimeMillis() / 1000) + MAX_TIMESTAMP_FUTURE;
         }
         return transactionViewModel.getAttachmentTimestamp() < snapshotTimestampMs
                 || transactionViewModel.getAttachmentTimestamp() > System.currentTimeMillis() + MAX_TIMESTAMP_FUTURE_MS;
     }
 
+    /**
+     * Runs the following validation checks on a transaction:
```

Comment:
because validation is implicit, it should be mentioned - that if invalid an exception is thrown.

Reply:
I didn't put any because of the `@throws`, but I guess I can add to the description